app/models/organization.rb

```diff
@@ -2,68 +2,70 @@
 require 'json'
 
 class Organization
+  attr_accessor :id, :name, :description, :imageUrl, :categories, :interests
 
-    attr_accessor :id, :name, :description, :imageUrl, :categories, :interests
+  def initialize(init_params)
+    @id = init_params['organizationId']
+    @name = init_params['name']
+    @description = init_params['description']
+    @imageUrl = init_params['profileImageUrl']
+    @categories = init_params['type']
+    @interests = init_params['categories']
+  end
 
-    def initialize(id, name, description, imageUrl, categories, interests)
-        @id = id
-        @name = name
-        @description = description
-        @imageUrl = imageUrl
-        @categories = categories
-        @interests = interests
-    end
+  def self.get_organizations(num, categories, interests)
+    orgs = self.get_orgs_from_api()
 
-    def self.get_organizations(num, categories, interests)
-	
-        orgs = self.get_orgs_from_api()
-	
-        top_orgs = []
-        orgs.each do |i|
-            if categories == nil || categories.size == 0 || categories.any?{ |s| s.casecmp(i['type']) == 0 }
-                top_orgs.push(Organization.new(i['organizationId'], i['name'], i['description'], i['profileImageUrl'], i['type'], i['categories']))
-            end
-        end
-	
-        if interests != nil
-            top_orgs = top_orgs.sort_by { |org| -org.count_interest(interests)}
-        end
-	
-        return top_orgs[0 ... num]
-    end
-  
-    def self.get_orgs_from_api()
-        url = 'https://callink.berkeley.edu/api/Organizations'
-        orgs = []
-        pageNumber = 0
-	
-        loop do
-            pageNumber += 1
-            begin
-                response = RestClient.get url, :params => {:page => pageNumber, :key => Figaro.env.callink_key}
-                parsed = JSON.parse(response)
-            rescue RestClient::ExceptionWithResponse => err # We don't have the keys yet, so on 401 error display placeholder orgs
-                placeholder = File.new("spec/response/raw_organizations_response.txt")
-                parsed = JSON.parse(placeholder.read)
-            end
-            orgs += parsed['items']
-            totalPages = parsed['totalPages']
-            pageNumber = parsed['pageNumber']
-	  
-            break if pageNumber >= totalPages
+    top_orgs = []
+    orgs.each do |i|
+      org_categories = i['categories']
+      org_categories.each do |c|
+        categories.each do |quiz_category|
+          if c['categoryName'].downcase.include? quiz_category.downcase
+            top_orgs.push(Organization.new(i))

```

Comment:
Slightly worrying -- I know that each org should only have 1 matching category, but if it ever doesn't then it's possible that it gets added twice. I would suggest using a break but it's two loops in. Maybe refactor?

Reply:
This is something I could fix while working on my part also, so not blocker for pr approval.

Reply:
I did this because the excel sheet with all the categories provided to us used key terms as opposed to the whole string as it appears in the database. I had to use this workaround because the previous code was not finding any orgs. Feel free to make changes as you see fit though, 100% agree this many for loops is clunky and can cause problems!